server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -325,6 +327,60 @@ class DeltaSharingService(serverConfig: ServerConfig) {
     streamingOutput(Some(queryResult.version), queryResult.responseFormat, queryResult.actions)
   }
 
+
+  @Post("/shares/{share}/schemas/{schema}/tables/{table}/queries/{queryId}")
+  @ConsumesJson
+  def getQueryStatus(
+     req: HttpRequest,
+     @Param("share") share: String,
+     @Param("schema") schema: String,
+     @Param("table") table: String,
+     @Param("queryId") queryId: String,
+     request: GetQueryInfoRequest): HttpResponse = processRequest {
+
+    if (table == "tableWithAsyncQueryError") {
+      throw new DeltaSharingIllegalArgumentException("expected error")
+    }
+
+    if(rand.nextInt(100) > 50) {
```

Comment:
Could you add comment for this?

Reply:
why do we need this 50% randomness? 
Is this covered by any tests?

Reply:
it suppose that the client should cover both cases.

Reply:
@linzhou-db I had made this suggestion.
The idea is that the server will return that query is still in progress sometimes, which will test the async code path.
@jadewang-db If there is a deterministic way of doing this (maybe return kPending for the first few calls, that is fine as well).

Reply:
sorry for the delay on this...

It seems to me that returning "pending" for the first few(3) calls is better than this.

Reply:
easist way to do is to make the first call pending

Reply:
@linzhou-db do you feel strong on this? I think this is the best way of returning some pending status, and this is not production code, I think we should be ok here?